### PR TITLE
Deep dive 11-private-content-on-ipfs

### PR DESCRIPTION
**Pull request by ianopolous**
I've made a suggestion for one possible direction this could take. 
@alanshaw Let me know what you think?

### CODE REVIEW

File: DEEP_DIVES/11-private-content-on-ipfs.md [hunk 1]
```diff
@@ -1,15 +1,23 @@
 # Private content on IPFS
+Can we have a private implementation of a group chat on IPFS?
```
Comment (alanshaw) [inline DEEP_DIVES/11-private-content-on-ipfs.md]:
```suggestion
Can we have a private group chat implemented on IPFS?
```

File: DEEP_DIVES/11-private-content-on-ipfs.md [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
-<Define what we would get if the Open Problem(s) are solved for this Deep Dive>
+See if we can use MLS directly in a decentralized setting. Possibly using Peergos primitives. 
 
 ## Requirements to consider
 
-<Do we have requirements (i.e. work with 1MM nodes)?>
+Same as in the MLS requirements we should scale to 50,000 member groups.
 
 ## Where to learn about it
+https://github.com/Peergos/Peergos/issues/427
```
Comment (alanshaw) [inline DEEP_DIVES/11-private-content-on-ipfs.md]: Can we get a link to TweetNaCl, Signal Protocol and Messaging Layer Security here as well please?

File: DEEP_DIVES/11-private-content-on-ipfs.md (hunk 1 above)
Comment (alanshaw) [reply DEEP_DIVES/11-private-content-on-ipfs.md]: Should this be rephrased as "private group file sharing"?

File: DEEP_DIVES/11-private-content-on-ipfs.md [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 
-<Short summary of topic including Open Problem(s) to solve>
+Motivation (goals - solutions):
+0. Don't expose content to the world on IPFS - encrypt before putting in IPFS
+1. Avoid having to manually manage keys - use password based key derivation
+2. Encrypt a single file - use a password based key derivation as in Minilock
+3. Encrypt entire directory trees, hiding metadata - use cryptree as in Peergos
+4. 1-1 Sharing - need asymmetric cryptography, Curve25519 (e.g. TweetNaCl), solved in Peergos
+5. 1-N sharing - Repeat 3 for each target. O(N) or O(1) for multiple shares with a proxy capability
+6. N-N sharing - hmmm. Essentially want a many-to-many version of Signal Protocol. People are working on such a thing, [Messaging Layer Security](https://github.com/mlswg/mls-architecture/blob/master/draft-ietf-mls-architecture.md) but it currently depends on a central reliable Delivery Service. Can we make this work in the decentralized setting? Can use Peergos primitives, e.g. sharing a file, and our PKI. Can it be a CRDT?
```
Comment (alanshaw) [inline DEEP_DIVES/11-private-content-on-ipfs.md]: Interesting - I like how you've covered the history of solved problems here as well. May be worth revisiting on the day to see if different ideas come to light or if new tech exists to solve them better!

File: DEEP_DIVES/11-private-content-on-ipfs.md [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
 
 ## Goal
 
-<Define what we would get if the Open Problem(s) are solved for this Deep Dive>
+See if we can use MLS directly in a decentralized setting. Possibly using Peergos primitives.
```
Comment (alanshaw) [inline DEEP_DIVES/11-private-content-on-ipfs.md]: Might be worth expanding (MLS -> Messaging Layer Security) and linking here so people skimming the document can still understand without having to read the full intro! 😁

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: DEEP_DIVES/11-private-content-on-ipfs.md (hunk 4 above)
Comment (ianopolous) [reply DEEP_DIVES/11-private-content-on-ipfs.md]: Done

File: DEEP_DIVES/11-private-content-on-ipfs.md (hunk 2 above)
Comment (ianopolous) [reply DEEP_DIVES/11-private-content-on-ipfs.md]: Done

**Pull request merged**